include/drivers/uart.h

```diff
@@ -1460,6 +1178,319 @@ static inline void uart_irq_callback_set(const struct device *dev,
 	uart_irq_callback_user_data_set(dev, cb, NULL);
 }
 
+/**
+ * @}
+ */
+
+/**
+ * @defgroup uart_async Async UART API
```

Comment:
can you add async events (enum + structures) to that group as well?

Reply:
Good point. Done.